applications/asset_tracker/src/cloud_codec/service_info.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include "service_info.h"
+
+#define SERVICE_INFO_JSON_NAME "serviceInfo"
+#define SENSORS_JSON_NAME "sensors"
+#define FOTAS_JSON_NAME "fota_v1"
+
+static int add_array_obj(const char * const items[], const u32_t item_cnt,
+			 const char * const item_name, cJSON * const obj)
+{
+	if ((items == NULL) || (item_name == NULL) || (obj == NULL)) {
+		return -EINVAL;
+	}
+
+	cJSON * array = cJSON_CreateArray();
+
+	if (array == NULL) {
+		return -ENOMEM;
+	}
+
+	for (u32_t cnt = 0; cnt < item_cnt; ++cnt) {
+		if (items[cnt] != NULL) {
+			cJSON_AddItemToArray(array,
+					     cJSON_CreateString(items[cnt]));
+		}
+	}
+
+	cJSON_AddItemToObject(obj, item_name, array);
+
+	return 0;
+}
+
+int service_info_json_object_encode(const char * const sensors[],
+				    const u32_t sensor_count,
+				    const char * const fota[],
+				    const u32_t fota_count,
+				    cJSON * const obj_out)
```

Comment:
Possible alignment issue here as well

Reply:
i initially ran clang-format on these files... that resulted in some CI errors about asterisk spacing 🤷‍♂

Reply:
Yeah, seems like clang-format, uncrustify and friends don't always do what checkpatch wants even with the config files in the repo

Reply:
The checkpatch complaint on the asterisk is weird, though. The style guide says it should be next to the variable name and not surrounded by spaces. Maybe it just checks for consistency within a patch or something